src/PowerLimiter.cpp

```diff
@@ -223,7 +223,21 @@ void PowerLimiterClass::setNewPowerLimit(std::shared_ptr<InverterAbstract> inver
             _lastCommandSent = millis();
         } 
         MessageOutput.printf("[PowerLimiterClass::loop] Limit Non-Persistent: %d W\r\n", newPowerLimit);
-        inverter->sendActivePowerControlRequest(Hoymiles.getRadio(), newPowerLimit, PowerLimitControlType::AbsolutNonPersistent);
+
+        std::list<ChannelNum_t> dcChnls = inverter->Statistics()->getChannelsByType(TYPE_DC);
+        int dcProdChnls = 0, dcTotalChnls = dcChnls.size();
+        for (auto& c : dcChnls) {
+            if (inverter->Statistics()->getChannelFieldValue(TYPE_DC, c, FLD_PDC) > 0) {
```

Comment:
I've unplugged the cables from one channel, and noticed that there is still some power being shown (`0.1 W`), even after 10 minutes, so the `> 0` actually didn't work for me. I therefore changed it to `> 1.0`, should be good enough.
It's maybe because the capacitor still has voltage on it, or I don't know...

```suggestion
            // There always seems to be some power < 1.0W, even when there are no cables connected.
            if (inverter->Statistics()->getChannelFieldValue(TYPE_DC, c, FLD_PDC) > 1.0) {
```

Reply:
We could also check for a voltage > 20V instead?

Reply:
@berni2288 Thanks for mentioning that issue. I applied your suggestion using `> 1.0` with a slight modification from my side to explicitly fall back to the `newPowerLimit` in such a case.